zipline/pipeline/loaders/events.py

```diff
@@ -148,7 +152,8 @@ def __init__(self,
                 raise ValueError(
                     WRONG_MANY_COL_DATA_FORMAT_ERROR.format(sid=k)
                 )
-
+        self.events_by_sid = {sid: df.dropna(subset=[self.event_date_col]) for
```

Comment:
I would have expected this filtering to hapen in `BlazeEventsLoader` at query time, rather than here.  Is there a reason we want the null rows to make it this far?